src/txn/txn_ckpt.c

```diff
@@ -405,6 +405,20 @@ __checkpoint_reduce_dirty_cache(WT_SESSION_IMPL *session)
 	 */
 	if (cache_size < 10 * WT_MEGABYTE)
 		return;
+
+	/*
+	 * Skip scrubbing if it won't perform at-least some minimum amount of
+	 * work. Scrubbing is supposed to bring down the dirty data to eviction
+	 * checkpoint target before the actual checkpoint starts. Do not perform
+	 * scrubbing if the dirty data to scrub is less than a pre-configured
+	 * size. This size is to an extent based on the configured cache size
+	 * without being too large or too small for large cache sizes.
+	 */
+	if ((__wt_cache_dirty_leaf_inuse(cache) +
+	    WT_MIN((0.2 * conn->cache_size) / 100, 500 * WT_MEGABYTE)) <
+	    ((cache->eviction_checkpoint_target * conn->cache_size) / 100))
```

Comment:
In parsing this, wouldn't we skip scrubbing in this case anyway?  The second half of the computation is the (default) 5% of cache for the checkpoint target.  So if the dirty cache is already less than the eviction_checkpoint_target we would (today) break out of the for loop a few lines below.

I thought the intent was to skip scrubbing if we're close to the 5%.  So shouldn't the calculation be something like:
if (wt_cache_dirty_leaf_inuse(cache) <
((eviction_checkpoint_target * cache_size)/100 + <calculation of meaning of "close to scrub target" >)

Reply:
You are right. I fixed the logic